packages/material-ui-lab/src/internal/pickers/PickersArrowSwitcher.tsx

```diff
@@ -84,16 +93,16 @@ const PickersArrowSwitcher = React.forwardRef(function PickersArrowSwitcher(
   const theme = useTheme();
   const isRtl = theme.direction === 'rtl';
 
-  const LeftArrowButton = components.LeftArrowButton || IconButton;
+  const LeftArrowButton = components.LeftArrowButton || PickersArrowSwitcherButton;
   const leftArrowButtonProps = componentsProps.leftArrowButton || {};
   const LeftArrowIcon = components.LeftArrowIcon || ArrowLeftIcon;
 
-  const RightArrowButton = components.RightArrowButton || IconButton;
+  const RightArrowButton = components.RightArrowButton || PickersArrowSwitcherButton;
```

Comment:
This looks off, shouldn't `visibility: hidden` be applied on  `components.RightArrowButton` too?

Reply:
It will be applied because of `styleProps` and Left, RightButton point to the same StyledComponent

```jsx
const PickersArrowSwitcherButton = styled(IconButton)
(({ styleProps = {} }) => ({
  ...(!!styleProps.hidden && {
    visibility: 'hidden',
  }),
}));

// render
const LeftArrowButton = components.LeftArrowButton || PickersArrowSwitcherButton;
const RightArrowButton = components.RightArrowButton || PickersArrowSwitcherButton;
<LeftArrowButton styleProps={{ ...leftArrowButtonProps, hidden: isLeftHidden }} />
<RightArrowButton styleProps={{ ...leftArrowButtonProps, hidden: isLeftHidden }} />
```
> code reduced for readability

Reply:
I think let's not consider that in this PR. at least the outside will receive the `styleProps` and `isLeftHidden, isRightHidden` are also come from props.

Reply:
I would say in order to preserve the previous behavior, we should use the `as` prop, instead of replacing the component. That way we would still have the original styles applied.

Reply:
In the future, the `components` props should work like your implementation, and developers should use the `as` prop for this particular usage, but it is differently implemented for the pickers right now.

For clarification there are two different usages:
- I want to replace the slot in the component with different one(will remove the styles too applied originally) - use `components` prop for this use-case
- I want to just change the element rendered - use `as` prop in this case for that slot

Reply:
Is this what you suggest?
```js
<PickersArrowSwitcherRoot ref={ref} {...other}>
  <PickersArrowSwitcherButton
    as={components.LeftArrowButton}
  />
  ...
  <PickersArrowSwitcherButton
    as={components.RightArrowButton}
  />
```

Reply:
okay, will make the change now. it is better.